api/agent/drivers/docker/docker_client.go

```diff
@@ -262,13 +363,15 @@ func (d *dockerWrap) PullImage(opts docker.PullImageOptions, auth docker.AuthCon
 func (d *dockerWrap) RemoveContainer(opts docker.RemoveContainerOptions) (err error) {
 	// extract the span, but do not keep the context, since the enclosing context
 	// may be timed out, and we still want to remove the container. TODO in caller? who cares?
-	span, _ := opentracing.StartSpanFromContext(opts.Context, "docker_remove_container")
-	defer span.Finish()
-	ctx := opentracing.ContextWithSpan(context.Background(), span)
+	ctx := common.BackgroundContext(opts.Context)
+	ctx, span := trace.StartSpan(ctx, "docker_remove_container")
+	defer span.End()
 
 	ctx, cancel := context.WithTimeout(ctx, retryTimeout)
 	defer cancel()
 
+	opts.Context = ctx
```

Comment:
none of the callers of this function set a timeout in agent.. Now, we start to impose this and removals can timeout.
I don't think we should set opts.Context with the new ctx.

I think we can revisit these retry timeouts in general in this package in the future. In some functions, ctx arg is used only for logging, in some it's used to enforce a timeout, sometimes retry timeout can hold up the request longer, could be using docker or dockeNoTimeout client, etc.

Reply:
thanks, this was a bug